File: genetic.pas
(***************************************************************)
(*            Fifteenth Annual UCF High School                 *)
(*                 Programming Tournament                      *)
(*                      May 4, 2001                            *)
(*                                                             *)
(*    Problem: Genetic                                         *)
(*    Written by: Phillip Dexheimer                            *)
(*    Solved by: Carlos Rosas-Anderson                         *)
(***************************************************************)
program Genetic;

var
	InputFile: Text;
	I, N, B: Integer;
	Gene: string;

begin
	{Associate the filename with the file variable}
	Assign(InputFile, 'GENETIC.IN');

	{Open the intput file for reading}
	Reset(InputFile);

	{Read the number of cases from the file}
	Readln(InputFile, N);

	{Process each of the N cases}
	for I := 1 to N do
	begin
		{Read the bit to be mutated from the file}
		Readln(InputFile, B);

		{Read the gene from the file}
		Readln(InputFile, Gene);

		{Mutate the bit}
		if Gene[B] = '1' then
			Gene[B] := '0'
		else
			Gene[B] := '1';

		{Print the gene}
		Writeln(Gene);
		Writeln;
	end;

	{Close the input file}
	Close(InputFile)
end.

File: genetic.c
/***************************************************************/
/*            Fifteenth Annual UCF High School                 */
/*                 Programming Tournament                      */
/*                      May 4, 2001                            */
/*                                                             */
/*    Problem: Genetic                                         */
/*    Written by: Phillip Dexheimer                            */
/*    Solved by: Carlos Rosas-Anderson                         */
/***************************************************************/
#include <stdio.h>

int main()
{
	FILE *file;
	int  i, n, b;
	char gene[30];

	
	// Open the input file for reading
	file = fopen("genetic.in", "r");

	// Read the number of cases from the file
	fscanf(file, "%d\n", &n);

	// Process each of the n cases
	for(i = 0; i < n; i++)
	{
		// Read the bit to be mutated
		fscanf(file, "%d\n", &b);

		// Read the gene
		fscanf(file, "%s\n", gene);

		// Mutate the bit
		if(gene[b - 1] == '1')
			gene[b - 1] = '0';
		else
			gene[b - 1] = '1';

		// Print out the mutated gene
		printf("%s\n\n", gene);
	}

	// Close the input file
	fclose(file);

	return 0;
}
